files/bin/raid-local-disks

```diff
@@ -0,0 +1,92 @@
+#!/usr/bin/env bash
+
+set -o errexit
+set -o pipefail
+set -o nounset
+
+# Sets up a RAID-0 of NVMe instance storage disks, moves
+# the contents of /var/lib/kubelet and /var/lib/containerd
+# to the new mounted RAID, and symlinks the kubelet and
+# containerd state directories.
+
+disks=($(find -L /dev/disk/by-id/ -xtype l -name '*NVMe_Instance_Storage_*'))
+## Bail early if there are no ephemeral disks to setup
+if [[ "${#disks[@]}" -eq 0 ]]; then
+  echo "no ephemeral disks found, skipping raid setup"
+  exit 0
+fi
+
+## Get devices of NVMe instance storage ephemeral disks to RAID
+EPHEMERAL_DISKS=($(realpath "${disks[@]}" | sort | uniq))
+
+MD_NAME="kubernetes"
+MD_DEVICE="/dev/md/${MD_NAME}"
+MD_CONFIG="/.aws/mdadm.conf"
+MOUNT_POINT="/mnt/${MD_NAME}"
+mkdir -p "$(dirname "${MD_CONFIG}")"
+
+if [[ ! -s "${MD_CONFIG}" ]]; then
+  mdadm --create --force --verbose \
+    "${MD_DEVICE}" \
+    --level=0 \
+    --name="${MD_NAME}" \
+    --raid-devices="${#EPHEMERAL_DISKS[@]}" \
+    "${EPHEMERAL_DISKS[@]}"
+  mdadm --detail --scan > "${MD_CONFIG}"
+fi
+
+# Format the array if not already formatted.
+if ! blkid --match-token TYPE=xfs "${MD_DEVICE}"; then
+  ## By default, mkfs tries to use the stripe unit of the array (512k),
+  ## for the log stripe unit, but the max log stripe unit is 256k.
+  ## So instead, we use 32k (8 blocks) to avoid a warning of breaching the max.
+  ## mkfs.xfs defaults to 32k after logging the warning since the default log buffer size is 32k.
+  mkfs.xfs -l su=8b "${MD_DEVICE}"
```

Comment:
Have you tested this on a filesystem that was already formatted with, say, ext4? The `if` condition will resolve to `true` but this might fail.